src/register.js

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from 'react';
+import { addons } from '@storybook/addons';
+import { useChannel } from '@storybook/api';
+// import { styled, Global, Theme, withTheme } from '@storybook/theming';
+import { Button, Separator } from '@storybook/components';
+import { getArrayItemByValue, getEmFromPx, getPxFromEm, getRandomVal } from './utils';
+import constants from './constants';
+
+const {
+  addonId,
+  panelId,
+  viewModes,
+  presets,
+  wrapperWidthChangeEvent,
+  wrapperMaxWidthChangeEvent,
+  toolWidthChangeEvent
+} = constants;
+
+const TaffyTool = () => {
+  const [toolPxWidth, setToolPxWidth] = useState('');
+  const [maxWidth, setmaxWidth] = useState(0);
+  const [presetWidths, setPresetWidths] = useState(presets);
+
+  const handleWrapperWidthChange = width => {
+    setToolPxWidthAsString(width);
+  };
+
+  const handleMaxWidthChange = width => {
+    const largePreset = getArrayItemByValue(presetWidths, 'large');
+    if (largePreset) {
+      const modifiedLargePreset = {
+        ...largePreset,
+        max: width
+      };
+      const newPresetWidths = presetWidths.map(obj => modifiedLargePreset.id === obj.id ? modifiedLargePreset : obj);
+      setPresetWidths(newPresetWidths);
+      setmaxWidth(width);
+    }
+  };
+
+  const handleInputChange = width => {
+    setToolPxWidth(width);
+  };
+
+  const handleInputBlur = width => {
+    setToolPxWidth(width);
+    emit(toolWidthChangeEvent, width);
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    setToolPxWidthAsString(toolPxWidth);
+    emit(toolWidthChangeEvent, toolPxWidth);
+  };
+
+  const handlePresetClick = presetId => {
+    const preset = getArrayItemByValue(presetWidths, presetId);
+    const randomWidthWithinRange = getRandomVal(preset.min, preset.max);
+    setToolPxWidthAsString(randomWidthWithinRange);
+    emit(toolWidthChangeEvent, randomWidthWithinRange);
+  };
+
+  const handleFullClick = () => {
+    setToolPxWidthAsString(maxWidth);
+    emit(toolWidthChangeEvent, maxWidth);
+  };
+
+  const emit = useChannel({
+    [wrapperWidthChangeEvent]: handleWrapperWidthChange,
+    [wrapperMaxWidthChangeEvent]: handleMaxWidthChange
+  });
+
+  const setToolPxWidthAsString = width => setToolPxWidth(width.toString());
+
+  return (
+    <>
+      <Separator />
+      <form onSubmit={handleSubmit} style={{
+        alignItems: 'center',
+        display: 'flex'
+      }}>
+        <div className="taffy-field">
+          <input
```

Comment:
[suggestion] Not sure if this will mess up your input styling/appearance or not, but if this input was of type `number`, we'd get the increment/decrement interface and be able to use the keyboard up/down arrows to change the value.

Reply:
I went back and forth on this and settled on `text` so I could limit maxLength (I know I can do that with JS on a number but didn't feel like it), but on second thought, I think the added input validation and increment/decrement interface and keyboard accessibility make it worth switching. I'm not going to restrict max length for right now, but that could be a future enhancement.